cmd/tailscale/cli/exitnode.go

```diff
@@ -70,10 +70,16 @@ func runExitNodeList(ctx context.Context, args []string) error {
 			// We only show exit nodes under the exit-node subcommand.
 			continue
 		}
-
 		peers = append(peers, ps)
 	}
 
+	// filterFormatAndSortExitNodes does not define a strict total order.
+	// As a quick fix, sort the input peers first so the printed out
+	// is deterministic between runs.
+	slices.SortFunc(peers, func(a, b *ipnstate.PeerStatus) int {
+		return strings.Compare(a.DNSName, b.DNSName)
+	})
+
```

Comment:
I think this should be in `filterFormatAndSortExitNodes()` at the top, since that is what the function is intended to do, and I'd prefer that the logic was kept in the same space. Also can drop the quick fix comment.